code/game/machinery/doors/airlock.dm

```diff
@@ -1378,6 +1380,18 @@ GLOBAL_LIST_EMPTY(airlock_emissive_underlays)
 		emagged = TRUE
 		return 1
 
+/obj/machinery/door/airlock/cmag_act(mob/user)
+	if(!operating && density && arePowerSystemsOn() && !cmagged)
+		operating = TRUE
+		update_icon(AIRLOCK_EMAG, 1)
+		sleep(6)
+		if(QDELETED(src))
+			return
+		operating = FALSE
+		update_icon(AIRLOCK_CLOSED, 1)
+		cmagged = TRUE
+		return 1
+
```

Comment:
```suggestion
/obj/machinery/door/airlock/cmag_act(mob/user)
	if(operating || cmagged || !density || !arePowerSystemsOn())
		return
	operating = TRUE
	update_icon(AIRLOCK_EMAG, 1)
	sleep(6)
	if(QDELETED(src))
		return
	operating = FALSE
	update_icon(AIRLOCK_CLOSED, 1)
	cmagged = TRUE
	return TRUE
```
early returns are preferred, could you double check the logic I recommended and make sure it's originally what you wanted?